### ISSUE TITLE
Exclude snap from Disk's to allow seeing the real disks

### ISSUE THREAD

**Issue by Darkyere** [ClementTsang/bottom#431]
### Describe the feature request

A startup flag to the command btm or similar that exlude all snap loop devices from the disk menu.

### Additional context/details

This is because i never see anything else than snap loop devices in Disk's list and would find it more interesting to see my actual HDD's and such

**Comment by ClementTsang** [comment]
Would using the [disk filter functionality](https://github.com/ClementTsang/bottom#disk-temperature-and-network-filtering) be enough for now?  Ideally it would be easier to configure in-app (another thing on the ever-growing backburner), but assuming the disk names are like `/dev/loop`, off the top of my  head, you could do:

```toml
[disk_filter]
is_list_ignored = true
list = ["/dev/loop.*"]
regex = true
```
or something like that.  I intend on adding a filter based on mounts in the future as well, if that's required.

**Comment by Darkyere** [comment]
I belive it would suffice.

I just cant figure out how to do that on a snap.

Any clue of how i should proceed?

**Comment by ClementTsang** [comment]
Could you send a screenshot of what your disk widget looks like?

**Comment by Darkyere** [comment]
This is how my widget looks like

![Ubuntu 20 04 1 bottom](https://user-images.githubusercontent.com/23291749/110295723-b70f2100-7ff1-11eb-844e-8ca8b1ee30d4.png)

**Comment by ClementTsang** [comment]
Yeah, I think something like:

```toml
[disk_filter]
is_list_ignored = true
list = ["/dev/loop.*"]
regex = true
```

inside the bottom.toml config file would work for your case.

**Comment by Darkyere** [comment]
That helped up a lot removed like 12 loop devices from snap.

----

Is it possible to do the same for mount paths in disk_filter?

I now only got 2 disk in that are snap disks.

But unfortunately they aren't called loop devices but the same as what the system is using so i cant really removed them like with /dev/loop

Luckily in the disk section their mount paths are /var/lib/snapd

Is it possible to filter them out by mount paths as well ?

Ty btw for getting me so far

**Comment by ClementTsang** [comment]
Glad to hear that worked.

As for mount paths, currently, no, that isn't available yet (shouldn't be hard to add but I'm a bit busy right now... school sucks).  I will definitely try to add it soon though.

Not 100% sure if this will work for your case but for now, would just adding the disks directly to the ignore list work?  For example:
```toml
list = ["/dev/loop.*", "disk_to_ignore", "other_disk_to_ignore"]
```

**Comment by Darkyere** [comment]
Unfortunately not.

Its like saying that `/dev/sda2` is `/` and the `/var/lib/snapd` is allso `/dev/sda2`
So i would just be removing my main drive aswell :D

But ty for getting me this far.
With 12 loop devices gone i can finally see my main drives, which is the most important part i figure.

Looking forward for the mount filter option i future releases.

Best regards,
Darkyere

**Comment by ClementTsang** [comment]
Just released 0.6.0, which has a mount filter option.

Something like:
```
[mount_filter]
is_list_ignored = true
list = ["/var/lib/snapd.*"]
regex = true
```
should be what you want.  Let me know if you have any issues.

**Comment by Darkyere** [comment]
This works perfectly thank you.

Was a bit slow about responding,
because i didn't have the time to look at it.

This is a great little enhancement,
Allows people to filter out disk in aany way needed.

Thank you and best regards,
Darkyere
